### PR TITLE
feat(subnavigation): add subnavigation selector

### PR DESCRIPTION
**Pull request by rosalie-liu**
### Proposed Changes

add subnavigation selector

### Potential Breaking Changes

nope

### Acceptance Criteria

-   [x ] The proposed changes are covered by unit tests
-   [x ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/react-vapor/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/react-vapor/src/components/subNavigation/SubNavigationSelector.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import * as _ from 'underscore';
+
+import {IReactVaporState} from '../../ReactVapor';
+import {ISubNavigationState} from './SubNavigationReducers';
+
+const getItem = (state: IReactVaporState, subNavigationId: string): ISubNavigationState =>
```
Comment (gdostie) [inline packages/react-vapor/src/components/subNavigation/SubNavigationSelector.ts]:
```suggestion
const getSubNavigation = (state: IReactVaporState, subNavigationId: string): ISubNavigationState =>
```

File: packages/react-vapor/src/components/subNavigation/SubNavigationSelector.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import * as _ from 'underscore';
+
+import {IReactVaporState} from '../../ReactVapor';
+import {ISubNavigationState} from './SubNavigationReducers';
+
+const getItem = (state: IReactVaporState, subNavigationId: string): ISubNavigationState =>
+    _.findWhere(state.subNavigations, {id: subNavigationId});
+
+const getItemSelected = (state: IReactVaporState, subNavigationId: string): string =>
+    getItem(state, subNavigationId)?.selected;
```
Comment (gdostie) [inline packages/react-vapor/src/components/subNavigation/SubNavigationSelector.ts]: We could memoize the result with reselect's createSelector function

```ts
const getSelectedItem = createSelector(getSubNavigation, (subNav) => subNav?.selected);
```

Comment (pochretien) [inline packages/react-vapor/src/components/subNavigation/SubNavigationSelector.ts]:
```suggestion
    getItem(state, subNavigationId)?.selected ?? '';
```

File: packages/react-vapor/src/components/subNavigation/SubNavigationConnected.tsx [hunk 3]
```diff
@@ -11,14 +11,11 @@ import {
...
+import {SubNavigationSelector} from './SubNavigationSelector';
 
-const mapStateToProps = (state: IReactVaporState, ownProps: ISubNavigationOwnProps): ISubNavigationStateProps => {
-    const item: ISubNavigationState = _.findWhere(state.subNavigations, {id: ownProps.id});
-    return {
-        selected: item ? item.selected : '',
-    };
-};
+const mapStateToProps = (state: IReactVaporState, ownProps: ISubNavigationOwnProps): ISubNavigationStateProps => ({
+    selected: SubNavigationSelector.getItemSelected(state, ownProps.id) || '',
```
Comment (pochretien) [inline packages/react-vapor/src/components/subNavigation/SubNavigationConnected.tsx]:
```suggestion
    selected: SubNavigationSelector.getItemSelected(state, ownProps.id),
```

File: packages/react-vapor/src/components/subNavigation/SubNavigationSelector.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import {createSelector} from 'reselect';
+import * as _ from 'underscore';
+
+import {IReactVaporState} from '../../ReactVapor';
+import {ISubNavigationState} from './SubNavigationReducers';
+
+const getSubNavigation = (state: IReactVaporState, subNavigationId: string): ISubNavigationState =>
+    _.findWhere(state.subNavigations, {id: subNavigationId});
+
+const getSelectedItem = createSelector(getSubNavigation, (subNav) => subNav?.selected ?? '');
```
Comment (dmgauthier) [inline packages/react-vapor/src/components/subNavigation/SubNavigationSelector.ts]: ` (subNav) => subNav?.selected ?? '' ` so convinent

**Pull request merged**